File: array/numappearingonce.cpp
#include<vector>
using namespace std;
int getSingleElement(vector<int> &arr){
	int maxi=0;
	for(int i=0;i<arr.size();i++){
		maxi=max(maxi,arr[i]);
	}
	vector<int>hash(maxi+1,0);
	for(int i=0;i<arr.size();i++){
		hash[arr[i]]++;
	}
	for(int i=0;i<arr.size();i++){
		if(hash[arr[i]]==1){
			return arr[i];
		}
	}

	return -1;
}

File: array/numappearingonce.py
from typing import *

def getSingleElement(arr : List[int]) -> int:
    # Write your code here.
    maxi=max(arr)
    n=len(arr)
    hash = [0] * (maxi + 1)
    for i in range(n):
        hash[arr[i]]+=1
    for i in range(n):
        if hash[arr[i]]==1:
            return arr[i]
    
